### PR TITLE
added retrieval of auth token for vetext and added to header

### PR DESCRIPTION
**Pull request by jakehova**
[923](https://github.com/department-of-veterans-affairs/notification-api/issues/923)

Hotfix to apply VeText auth token for post authentication to VeText endpoint.

### CODE REVIEW

**Review by k-macmillan — DISMISSED** [review]

**Comment by k-macmillan** [comment]
Seems I jumped the gun a tad, unit tests are failing. Should have known with the new SSM lookup. Actual tests with VeText pass though:
image

**Review by cris-oddball — APPROVED** [review]

**Review by k-macmillan — APPROVED** [review]

**Review by kalbfled — CHANGES_REQUESTED** [review]
I won't hold you up for the get_parameters change due to our time crunch, but please open a tech debt ticket if you don't change that for this PR.

File: lambda_functions/two_way_sms/two_way_sms_v2.py [hunk 1]
```diff
@@ -67,16 +70,26 @@
...
-    sqlalchemy_database_uri = ssm_response.get("Parameter", {}).get("Value")
+    sqlalchemy_database_uri = ssm_database_uri_path_response.get("Parameter", {}).get("Value")
 
     if sqlalchemy_database_uri is None:
         sys.exit("Can't get the database URI from SSM Parameter Store.")
 
+    ssm_vetext_authtoken_path_response: dict = ssm_client.get_parameter(
+        Name=VETEXT_API_AUTH_SSM_PATH,
+        WithDecryption=True
+    )
```
[the remark this answers is not in the document]
Comment (k-macmillan) [reply lambda_functions/two_way_sms/two_way_sms_v2.py:86]: Great point and good to keep in mind to anyone looking at this. 
Given this a temp measure and only ran during execution environment (not invocations) I'm okay leaving it until the cleanup, but this is definitely something we need to be very aware of. `get_parameters` can only be called so many times per second due to AWS quotas.

File: lambda_functions/two_way_sms/two_way_sms_v2.py [hunk 2]
```diff
@@ -268,7 +281,8 @@ def forward_to_service(inbound_sms: dict, url: str) -> bool:
         return False
 
     headers = {
-        'Content-type': 'application/json'
+        'Content-type': 'application/json',
+        'Authorization': 'Basic ' + vetext_auth_token
```
[the remark this answers is not in the document]
Comment (k-macmillan) [reply lambda_functions/two_way_sms/two_way_sms_v2.py:285]: It does

File: lambda_functions/two_way_sms/two_way_sms_v2.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cris-oddball) [reply lambda_functions/two_way_sms/two_way_sms_v2.py:86]: @k-macmillan if this cleanup will happen in this sprint, I'd like for it to be changed now.  I really don't want to retest this twice before we deploy it to production.

[the remark this answers is not in the document]
Comment (jakehova) [reply lambda_functions/two_way_sms/two_way_sms_v2.py:86]: will hold off on this add this to a tech debt ticket

File: lambda_functions/two_way_sms/two_way_sms_v2.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jakehova) [reply lambda_functions/two_way_sms/two_way_sms_v2.py:285]: Yes.  I verified that the function works when deployed to dev and dev param store holding the same value as staging counterpart.

File: lambda_functions/two_way_sms/two_way_sms_v2.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (k-macmillan) [reply lambda_functions/two_way_sms/two_way_sms_v2.py:86]: The change to `get_parameters` is on indefinite hold.

[the remark this answers is not in the document]
Comment (jakehova) [reply lambda_functions/two_way_sms/two_way_sms_v2.py:86]: Ideally, we wont be pulling this in the future.  There needs to be another way to handle communication with their endpoint that will be standard about our comms with every endpoint.

**Pull request merged**